### ISSUE TITLE
Better Documentation (Jekyll / github pages)?

### ISSUE THREAD

**Issue by shafr** [vulhub/vulhub#179]
Has anyone come with idea of updating this project with Jekyll formatting ? 
* It would look much better
* universal format would be introduced (with front matter containing list of CVE's, server name, articles)
* search, index, tags would be introduced
* rendered site would be available by default on vulhub.github.io
* different layouts 
* use of templates and headers / footers ( for dockerfile, CVE introduction, etc)

What do you think about it ?

**Comment by shafr** [comment]
Example (theme for jekyll):
https://github.com/mmistakes/minimal-mistakes

**Comment by phith0n** [comment]
I would say something about official page of Vulhub:

- The official website of Vulhub is here <https://vulhub.org/> now, it is still Chinese, welcome to translate it too.
- The source code of this page is here <https://github.com/vulhub/vulhub-org>. the code is developed ourselves, which is a simple static site.
- Now the page is hosted on <https://www.netlify.com/>, which is also a static website hosting provider like GitHub pages.
- I think it's is better to use Jekyll or other well-known frameworks than a self-development project.
- What we need is a landing page rather than a blog, so can you recommend some themes that is fit for a open source project?

**Comment by shafr** [comment]
> What we need is a landing page + a document system rather than a blog, so can you recommend some themes that are fit for an open-source project?
What you want are not blog posts, but `collections` (see [link](https://jekyllrb.com/docs/collections/)). You can completelly ignore blog section. 
Regarding landing page - it's just matter of putting correct [layout](https://jekyllrb.com/docs/layouts/).


> The official site of Vulhub has all of your requirements such as search, index, tag https://vulhub.org/#/environments/, so if there is no suitable framework, just leave it as it is.
It's also very easy to display posts by tag, category, etc there using layouts.

Let me fork this project and create some demo as soon as I will find some time.

**Comment by phith0n** [comment]
Thanks, I'm not familiar with Jekyll, welcome to help me improve the project!

**Comment by shafr** [comment]
I see 2 problems that needs to be addressed: 
* Assets storage. By default structure of Jekyll posts requires to have: collection of markdown files with posts and folder Assets that would contain assets for corresponding folder:
```
├── assets
│   └── cve-01-test
│       ├── dockercompose.yaml
│       ├── img01.jpg
│       └── img02.jpg
└── _cves
    └── cve-01-test.md
```

This would render page cve-01-test which would reference assets.
Is it possible that new entries would follow this structure ?

* I18N or localization is hard thing. I still haven't figured out the best way to implement reference to CN page, except by having 2 collections - one for English and one for Chinese that would use same resources folder:
```
├── _vulns_cn
│   └── apache
└── _vulns_en
    └── apache
```

On the top page there would be 2 header links - for english and for chinese cve's this way.

**Comment by phith0n** [comment]
For problem 1, it's hard to change the current project structure because of making Jekyll compatible. In my option, our project is portable and easy to use, is not only a vulnerabilities manual but also a tool, which would be pulled and executed by the user, it's different from a blog. 

You can enter the folder, then found everything here, Dockerfile, Docker-Compose File, manual, screenshots, exploits. So if we change the structure, it will affect the experience, make the user confused.

But it also an alternative, convert the current project to a Jekyll-compatible project before the static website generation, let CI do it.

**Comment by shafr** [comment]
OK, understood. 
I've made a POC on https://shafr.github.io/beautiful-jekyll 

What I've noticed - that almost every entry has `docker-compose` section, which can be moved to header. 

Implemented:
* Index for EN and CN vulns using templates

Issues
* Images are not working due to all resources / links poining to root url.

What would be great todo:
* Depending on tag docker_compose variable in each post make a note on how-to start docker for each CVE
* search by tags

I'm closing this issue for now - your points make sense and it would be hard to track everyone's commit to follow same structure.
